awx/main/tasks.py

```diff
@@ -1381,6 +1381,11 @@ def pre_run_hook(self, job, **kwargs):
             if job.is_isolated() is True:
                 pu_ig = pu_ig.controller
                 pu_en = settings.CLUSTER_HOST_ID
+            if job.project.status != 'successful':
+                raise RuntimeError(
+                    'Project for this job template is not successful. '
```

Comment:
Maybe:

> Project for this job template failed to sync.

If you don't know Tower API terminology, this sentence may not be super clear.

Reply:
I don't like the use of "sync" there, because in this case we avoid running the sync.

You're right, this is user-facing so what about

> Playbook can not be reliably obtained because its associated project is {status}. Please correct and update the project.

Everyone knows what a playbook is!

Reply:
> The Project associated with this Job Template failed to update.

😄 ?

I don't love mentioning "project status" if you're not familiar w/ the API's notion of objects having a "status".  UI-only people know this concept as a red/green dot.

I wouldn't normally be so picky, but I think it's pretty important to get this message this right, since it's likely one of the first errors a user might encounter on install cc @tvo318

Reply:
> The Project associated with this Job Template failed to update.

I agree that this is very user-friendly, but my problem here is that it's just not true.

what about:

> The Project associated with this Job Template had a failed update.

Reply:
"The project revision for this job template is unknown due to a failed update". ?